packages/story-editor/src/app/media/useUploadMedia.js

```diff
@@ -335,7 +347,12 @@ function useUploadMedia({
             onUploadProgress,
             onUploadError,
             onUploadSuccess,
-            additionalData,
+            additionalData:
+            {
+              batchPosition: files.length - 1 - index,
+              batchCount: files.length,
+              ...additionalData
```

Comment:
```suggestion
              ...additionalData
              batchPosition: files.length - 1 - index,
              batchCount: files.length,
```

Also, `additionalData` might not be `undefined`, so let's initialize it to an empty object in the function params.